### ISSUE TITLE
r2d2 database is locked

### ISSUE THREAD

**Issue by laa** [golemfactory/yagna#1090]
I periodically see the following errors in the log.

```
2021-02-27T15:23:56Z WARN  ya_provider::market::provider_market] Failed to unsubscribe offers from the market: HTTP error requesting DELETE http://127.0.0.1:7465/market-api/v1/offers/c92c57a35aef401e8dd68b1f1f4e3393-1d1e60e6e069b57a1faaa9986c170b3580e6d79429a5a80062fb2a38009c7c16: 410 Gone; msg: 'Can't unsubscribe expired Offer [c92c57a35aef401e8dd68b1f1f4e3393-1d1e60e6e069b57a1faaa9986c170b3580e6d79429a5a80062fb2a38009c7c16].'
[2021-02-27T15:23:56Z INFO  ya_market::matcher] Subscribed new Offer: [8a54b21e14484beebf0b40f352b63808-4fe6d92fcadd80e33ec1358039c448dfc5fe89833763fa656db9dc57f0ea8e7b] using identity: golem-cli [0xa47cf9f5e384ff56fa584f300028ad19dcaf8319]
[2021-02-27T15:23:56Z INFO  ya_provider::market::provider_market] Subscribed offer. Subscription id [8a54b21e14484beebf0b40f352b63808-4fe6d92fcadd80e33ec1358039c448dfc5fe89833763fa656db9dc57f0ea8e7b], preset [vm].
[2021-02-27T15:35:51Z ERROR r2d2] database is locked
[2021-02-27T15:54:21Z ERROR r2d2] database is locked
[2021-02-27T15:54:21Z ERROR r2d2] database is locked
[2021-02-27T16:03:51Z ERROR r2d2] database is locked
[2021-02-27T16:03:51Z ERROR r2d2] database is locked
[2021-02-27T16:04:51Z ERROR r2d2] database is locked
[2021-02-27T16:04:51Z ERROR r2d2] database is locked
[2021-02-27T16:14:21Z ERROR r2d2] database is locked
[2021-02-27T16:14:21Z ERROR r2d2] database is locked
```

Is it normal behaviour?

Version  information:

```
Version    0.6.0        
Commit     ed55e851   
Date       2021-02-15 
Build      113
```

If you need I can perform steps to provide additional information following the instructions.

**Comment by laa** [comment]
I can build a version with debug information if you provide me build instructions, and then send you a full stack trace when such an event happened if that helps. But I can not perform a full analysis of the issue. Unfortunatelly.

**Comment by tenninebt** [comment]
Having the same issue with (macOS Catalina 10.15.17) :
```
Version    0.6.1        
Commit     a282c32b   
Date       2021-03-08 
Build      125
```

**Comment by fsondej** [comment]
It happens to me too

```
golemsp 0.6.1 (a282c32b 2021-03-08 build #125)
Ubuntu 20.04.2 LTS
```

**Comment by tenninebt** [comment]
FYI, I updated to 0.6.3, can't replicate the bug. @laa @fsondej could you update and test again?
Thanks

**Comment by mfranciszkiewicz** [comment]
Fixed by #1082
